### ISSUE TITLE
unable to pull image

### ISSUE THREAD

**Issue by Deshke** [aws/amazon-vpc-cni-k8s#334]
```bash
 ~> docker pull 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.3.2
Error response from daemon: Get https://602401143452.dkr.ecr.us-west-2.amazonaws.com/v2/amazon-k8s-cni/manifests/v1.3.2: no basic auth credentials
 ~> curl -vks https://602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.3.2
*   Trying 52.35.81.238...
* TCP_NODELAY set
* Connected to 602401143452.dkr.ecr.us-west-2.amazonaws.com (52.35.81.238) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: CN=*.dkr.ecr.us-west-2.amazonaws.com
*  start date: Nov 12 00:00:00 2018 GMT
*  expire date: Oct 30 12:00:00 2019 GMT
*  issuer: C=US; O=Amazon; OU=Server CA 1B; CN=Amazon
*  SSL certificate verify ok.
> GET /amazon-k8s-cni:v1.3.2 HTTP/1.1
> Host: 602401143452.dkr.ecr.us-west-2.amazonaws.com
> User-Agent: curl/7.58.0
> Accept: */*
> 
< HTTP/1.1 401 Unauthorized
< Content-Type: text/plain; charset=utf-8
< Date: Wed, 27 Feb 2019 07:56:43 GMT
< Docker-Distribution-Api-Version: registry/2.0
< Www-Authenticate: Basic realm="https://602401143452.dkr.ecr.us-west-2.amazonaws.com/",service="ecr.amazonaws.com"
< Content-Length: 15
< Connection: keep-alive
< 
Not Authorized
* Connection #0 to host 602401143452.dkr.ecr.us-west-2.amazonaws.com left intact
``` 

via a coreOS instance in us-east-2, ubuntu ami in us-east-2/eu-west-1

**Comment by mogren** [comment]
Hi @Deshke ,

Have you authenticated against ECR? Does running `aws sts get-caller-identity` work in the shell for you?

**Comment by chiefy** [comment]
I'm having the same issue...

```
 Warning  Failed   21m (x4 over 22m)     kubelet, ip-10-XXX-XXX-16.ec2.internal  Failed to pull image "602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:1.3.2": rpc error: code = Unknown desc = Error response from daemon: manifest for 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:1.3.2 not found
```

```
[ec2-user@ip-10-205-183-16 ~]$ aws sts get-caller-identity
{
    "Account": "XXXXXXXXX",
    "UserId": "xxxxxx:i-xxxx",
    "Arn": "arn:aws:sts::XXXXXXXXXXXXXX:assumed-role/eks-worker-xxxxxxx-20190305182307970000000007/i-xxxxxxxxxxxxxxxxxxx"
}
```

**Comment by mogren** [comment]
@chiefy There is now a v in front of the release number, so `docker pull 602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.3.2` should work. 

(Not sure why the switch was made between 1.3.1 and v1.3.2)

**Comment by nckturner** [comment]
Both tags are now available, v1.3.2 and 1.3.2.

**Comment by CadenOf** [comment]
@mogren  hi , mogren. I have the same issue as yours, how does you remove "V2" in front of the release number ? 
my images was pulled by `kubelet`, what I write in pods.yaml was  `xxxxxx.dkr.ecr.xxxxx.amazonaws.com.cn/k8s-mirror:pause-amd64-3.0`, but `kubelet` seems to access `https://xxxx.dkr.ecr.xxxxx.amazonaws.com.cn/v2/k8s-mirror/manifests/pause-amd64-3.0` and got the Error: `Error response from daemon: Get https://xxxx.dkr.ecr.xxxx.amazonaws.com.cn/v2/k8s-mirror/manifests/pause-amd64-3.0: no basic auth credentials`
